internal/pkg/structfield/structfield.go

```diff
@@ -220,6 +224,11 @@ type Offsets struct {
 
 	// values is a map between version and offset value.
 	values map[verKey]offsetVersion
+
+	// singleOffset is the single offset in the values map.
+	// If there is only one offset, this will be that offset.
+	// Otherwise, this will be invalidOffset.
+	singleOffset uint64
```

Comment:
A type will likely be more useful here so the zero value is unset and a "special" offset doesn't need to be used. E.g.:

```go
type single struct {
	value uint64
	valid bool
}
```

note, the naming is not well thought through but just and example.